ocaml/parsing/jane_syntax_parsing.mli

```diff
@@ -185,17 +185,26 @@ module type AST = sig
         the embedding by the first argument.
 
         If the given syntax feature does not actually extend the given syntactic
-        category, returns [None]; this will be reported as an error. (For
-        example: There are no pattern comprehensions, so when building the
-        extended pattern AST, this function will return [None] if it spots an
-        embedding that claims to be from [Language_extension Comprehensions].)
+        category, returns [None].
     *)
+    -> fail_if_wrong_syntactic_category:bool
```

Comment:
The reason is that when we call `Case.of_ast` on a normal (say, unguarded) case whose rhs happens to use a Jane_syntax extension, we don't want `Case.of_ast` to fail (which it would absent this flag being set to false).

Let me know if this is confusing or if you have a better way to design this.

Reply:
As discussed, let's move this to the `AST_internal` module instead so we can co-locate this boolean flag with the reason why we need it.